### PR TITLE
salt minion contact method

### PR DESCRIPTION
**Pull request by Loquacity**
~**Merge https://github.com/uyuni-project/uyuni-docs/pull/573 first!!**~

# Description

Adds the Salt Minion Config File section. 

~Still TODO:~
~* Add to nav when other PRs merged.~ Done

# Target branches

Which product version does this PR apply to?

- [ ] Manager-4.0
- [ ] Manager-4.1
- [x] Manager-4.2
- [x] Uyuni

# Links

Partial https://github.com/SUSE/spacewalk/issues/12477

### CODE REVIEW

File: modules/client-configuration/pages/contact-methods-salt.adoc [hunk 1]
```diff
@@ -0,0 +1,29 @@
+[[contact-methods-salt]]
+= Salt Minion Configuration File
+
+The simplest contact method for Salt clients is to use the client's internal DNS to resolve the FQDN of the {productname} Server.
```
[the remark this answers is not in the document]
Comment (keichwa) [reply modules/client-configuration/pages/contact-methods-salt.adoc:4]: The DNS thing isn't the contact method, IMO.  Working DNS is a prerequisite if you use the FQDN in the cfg file.

**Review by keichwa — APPROVED** [review]
I think we must make clear that there is not need to configure clients manually.  Our default bootstrap methods will enter that configuration.

**Comment by Loquacity** [comment]
> I think we must make clear that there is not need to configure clients manually. Our default bootstrap methods will enter that configuration.

Yeah, Pau expressed surprise that this wasn't documented, but I think it's mostly because there's not really any reason to register them like this :woman_shrugging:

File: modules/client-configuration/pages/contact-methods-salt.adoc (hunk 1 above)
[the remark this answers is not in the document]
Comment (Loquacity) [reply modules/client-configuration/pages/contact-methods-salt.adoc]: Yeah, I really struggled with wording this. I was trying to put the emphasis on the client making the connection to the Server (as opposed to the Server finding the client).

[the remark this answers is not in the document]
Comment (Loquacity) [reply modules/client-configuration/pages/contact-methods-salt.adoc]: Updated

**Review by keichwa — APPROVED** [review]

File: modules/client-configuration/pages/contact-methods-salt.adoc [hunk 2]
```diff
@@ -1,8 +1,29 @@
 [[contact-methods-salt]]
-= Contact Methods for Salt Clients
+= Salt Minion Configuration File
 
-Salt clients can communicate with the {productname} Server using SSH.
+In most cases, Salt clients are registered accurately with the default bootstrap methods.
+However, you can use Salt to register the client to the {productname} Server manually by editing the Salt minion file on the client, and providing the fully qualified domain name (FQDN) of the server.
+This method operates using port 4505 outbound on the server and port 4506 inbound.
```
Comment (brejoc) [inline modules/client-configuration/pages/contact-methods-salt.adoc:6]: @Loquacity Sorry that I'm so late to the party, but this is not correct. Both connections are initiated by the minion to the master: https://docs.saltstack.com/en/getstarted/system/communication.html

The relevant quotes:
`Publisher | (port 4505) All Salt minions establish a persistent connection to the publisher port…`
and
`Request Server | (port 4506) Salt minions connect to the request server as needed to send results to the Salt master…`

Comment (Loquacity) [reply modules/client-configuration/pages/contact-methods-salt.adoc:6]: Thanks for letting me know! New PR here: https://github.com/uyuni-project/uyuni-docs/pull/639

**Pull request merged**